### PR TITLE
20 app page history museum 2

### PR DESCRIPTION
**Pull request by M3dzZ**

### CODE REVIEW

File: apps/mobile/src/services/visit-history-service/visit-history-service-impl.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        with: {
+          journey: {
+            with: {
+              museum: true, 
+            }
+          }
+        }
+      })
+
+      console.error('Result from database:', result)
```
Comment (Zalerys) [inline apps/mobile/src/services/visit-history-service/visit-history-service-impl.ts:20]: delete this console.error, it's not an error

File: packages/ui/src/components/button.tsx [hunk 2]
```diff
@@ -1,95 +1,86 @@
...
+import { withVariants } from "../utils/with-variants";
 
 interface ButtonProps {
-  backgroundColor?: string
-  buttonMd?: boolean
-  children?: string
-  disabled?: boolean
-  icon?: typeof Icon[keyof typeof Icon] | undefined
-  onClick?: GetProps<typeof TamaguiButton>['onPress']
+  backgroundColor?: string;
```
Comment (Zalerys) [inline packages/ui/src/components/button.tsx:10]: Need eslint cleane, no `;`

File: apps/mobile/package.json [hunk 3]
```diff
@@ -7,7 +7,7 @@
     "dev:build-dry": "mkdir -p node_modules/.prebuild-rn",
     "dev:build-transform": "babel src --out-dir node_modules/.prebuild-rn --extensions .ts,.js,.tsx,.jsx,.cjs,.mjs --watch",
     "dev:expo": "expo start --tunnel --dev-client",
-    "dev:agrume": "agrume --tunnel bore --watch",
+    "dev:agrume": "agrume --tunnel localtunnel --watch",
```
Comment (arthur-fontaine) [inline apps/mobile/package.json:10]: It should stay `bore`, not `localtunnel`, it was a temporary fix for your side only.

File: apps/mobile/src/screens/home/home.tsx [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+import { Text } from 'react-native'
+import { CameraView } from 'expo-camera'
 
-import { JoinCard } from './components/join-card'
-import { DefaultLayout } from '../../shared/layouts/default-layout'
 
-/**
- * The home screen of the application.
- */
 export function HomeScreen() {
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/home.tsx:10]: There is absolutely no reason to change the `HomeScreen`. The version you put is outdated of few months.

File: apps/mobile/src/screens/home/hooks/use-visit-data.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
+import { useAuth } from '@clerk/clerk-expo'
+import { createRoute } from 'agrume'
+import * as DI from 'diabolo'
+import { useEffect, useState } from 'react'
+
+import { visitHistoryService } from '../../../services/visit-history-service/visit-history-service'
+import { serverImpls } from '../../../shared/utils/server-impls'
+import type { VisitWithJourneyAndMuseum } from '../../../../../../packages/db/src/db'
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/hooks/use-visit-data.ts:8]: Why you import with relative path? The goal of the `packages` is folder is to avoid this for shared packages. You should just use `db`.

File: apps/mobile/src/screens/home/types/visit.ts [hunk 6]
```diff
@@ -0,0 +1,16 @@
+export interface VisitByUserIdSerialized {
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/types/visit.ts:1]: Why did you recreate a type here? Couldn't you use the types coming from `db` package?

File: apps/mobile/src/shared/layouts/default-layout.tsx [hunk 7]
```diff
@@ -1,21 +1,19 @@
-import React from 'react'
-import { Box } from 'ui'
+import React from "react";
```
Comment (arthur-fontaine) [inline apps/mobile/src/shared/layouts/default-layout.tsx:1]: There is a lot formatting/linting errors. Please run `pnpm lint --fix`.

File: apps/mobile/src/utils/db.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import Database from 'better-sqlite3'
```
Comment (arthur-fontaine) [inline apps/mobile/src/utils/db.ts:1]: Why there is a db here?

File: packages/db/src/db.ts [hunk 9]
```diff
@@ -11,3 +11,10 @@ export type Visit = typeof visits.$inferSelect
 export type JourneyStep = typeof journeySteps.$inferSelect
 export type Museum = typeof museums.$inferSelect
 export type Car = typeof cars.$inferSelect
+
+export type VisitWithJourneyAndMuseum = Visit & {
```
Comment (arthur-fontaine) [inline packages/db/src/db.ts:15]: This type is unnecessary. It should be located in the mobile directory because it is only used on mobile-side. It doesn't need to be shared. It's just pollution. Please move this in mobile folder.

File: packages/ui/src/components/card.tsx [hunk 10]
```diff
@@ -23,22 +23,25 @@ export const Card = withVariants<
...
       borderWidth: 0,
       cursor: 'pointer',
       flexDirection: 'column',
-      padding: '$card',
+      margin: 9,
+      overflow: 'hidden',
+      padding: 36,
+      marginVertical: 40,
+      marginHorizontal: 20,
+      gap: 10,
```
Comment (arthur-fontaine) [inline packages/ui/src/components/card.tsx:36]: Why did you change the style of the `Card` component? I think it was correctly style.

File: packages/ui/src/components/histories.tsx [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+    onClick: () => void;
+    text: string;
+  };
+}
+
+interface HistoryProps {
+  histories: ItemProps[];
+}
+
+export const Histories = withVariants<
```
Comment (arthur-fontaine) [inline packages/ui/src/components/histories.tsx:42]: There is no point to put `Histories` component in the `ui` package. `ui` package is useful to share the components used in dashboard and mobile app (like buttons, inputs, cards, etc.). This component is very specific to mobile so it shouldn't be there, neither use Tamagui. Please move this in mobile folder, and remove usage of Tamagui (instead use the `Box` component from `ui` and built-in React Native components)

File: packages/ui/src/components/typographies/title1.tsx [hunk 12]
```diff
@@ -27,7 +27,6 @@ export const Title1 = withVariants<
       borderWidth: 0,
       display: 'flex',
       flexDirection: 'row',
-      gap: 8,
```
Comment (arthur-fontaine) [inline packages/ui/src/components/typographies/title1.tsx:30]: Why did you removed this?

File: packages/ui/src/components/sheet.tsx [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+interface SheetProps {
+    children?: string
+    open?: boolean
+    title?: string
+  }
+
+/**
+ *
+ */
+export function Sheet({ children, open: _open = true, title }: SheetProps) {
```
Comment (arthur-fontaine) [inline packages/ui/src/components/sheet.tsx:15]: Maybe I'm mistaken, but I didn't see any usage of this. Why did you add this? If it is not used, remove it please.

File: packages/ui/src/index.tsx [hunk 14]
```diff
@@ -7,6 +7,8 @@ export * from './components/input'
 export * from './components/otp-input'
 export * from './components/separator'
 export * from './components/theme-provider'
+export * from './components/histories'
+export * from './components/sheet'
```
Comment (arthur-fontaine) [inline packages/ui/src/index.tsx:11]: If you remove the `Sheet` component, don't forget to remove this line.

File: packages/ui/src/tamagui/tokens.ts [hunk 15]
```diff
@@ -3,7 +3,6 @@ import { createTokens } from '@tamagui/core'
 export const tokens = createTokens({
   color: {
     beige: '#FCF0E3',
-    beigeLight: '#FFFBF5',
```
Comment (arthur-fontaine) [inline packages/ui/src/tamagui/tokens.ts:6]: Why did you remove this?

**Review by arthur-fontaine — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/home/components/histories.tsx [hunk 16]
```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { ChevronRight, History, Image } from '@tamagui/lucide-icons';
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/components/histories.tsx:2]: You should use icons from theui package

File: apps/mobile/src/screens/home/home.tsx [hunk 17]
```diff
@@ -1,15 +1,70 @@
...
+    isCameraOpen,
+    openCamera,
+    requestPermissionToUseCamera,
+  } = useScanController();
+
+
+  if (isCameraOpen && hasPermissionToUseCamera) {
+    return (
       <DefaultLayout>
-          <JoinCard />
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/home.tsx:12]: JoinCard should not be removed. There is no reason to change the Home, except to add your history component

File: apps/mobile/src/screens/home/home.tsx [hunk 18]
```diff
@@ -1,15 +1,70 @@
...
+            text: "Scanner un QR Code",
+          }}
+          icon={Icon.QrCode}
+          text="Vous vous apprêtez à visiter un musée ? Demandez à l’accueil si ils prennent en charge les Yvees afin de rendre votre visite plus attractive !"
+          title="Rejoignez une session"
+          variant="default"
+        />
+      </ThemeProvider>
+
+      {visit && visit.length > 0 && (
```
Comment (arthur-fontaine) [inline apps/mobile/src/screens/home/home.tsx:60]: This should be controlled by the history component. In the history component, if there is no visit, then return null.

File: packages/db/src/schema/visits.ts [hunk 19]
```diff
@@ -11,7 +11,8 @@ export const visits = sqliteTable('visits', withTimestamps({
   inProgress: integer('in_progress', { mode: 'boolean' }).notNull(),
   journeyId: integer('journey_id').notNull(),
   userId: integer('user_id').notNull(),
-}))
+}));
```
Comment (arthur-fontaine) [inline packages/db/src/schema/visits.ts:14]: No ";"

File: packages/ui/src/components/button.tsx [hunk 20]
```diff
@@ -93,4 +93,4 @@ export const Button = withVariants<
         </TamaguiButton>
     )
   },
-)
+)
```
Comment (arthur-fontaine) [inline packages/ui/src/components/button.tsx:96]: There is a diff here but it shouldn't. I think it's that there is no blank line at the end of the file.

File: packages/ui/src/components/card.tsx [hunk 21]
```diff
@@ -31,14 +31,12 @@ export const Card = withVariants<
...
     },
     default: {
-      overflow: 'hidden',
     },
   },
 )(
   // eslint-disable-next-line ts/naming-convention
   ({ variant }, { action, icon: Icon, text, title }: CardProps) => {
     const ICON_SHIFT = 24
-
```
Comment (arthur-fontaine) [inline packages/ui/src/components/card.tsx:41]: Why these changes?

File: packages/ui/src/components/card.tsx [hunk 22]
```diff
@@ -71,4 +69,4 @@ export const Card = withVariants<
         </TamaguiCard>
     )
   },
-)
+)
```
Comment (arthur-fontaine) [inline packages/ui/src/components/card.tsx:72]: Diff but shouldn't. It should be a blank line at end

File: apps/mobile/src/utils/db.ts (hunk 8 above)
Comment (arthur-fontaine) [inline apps/mobile/src/utils/db.ts:1]: Remove this unnecessary file.

**Review by arthur-fontaine — CHANGES_REQUESTED** [review]

**Review by arthur-fontaine — APPROVED** [review]
